File: bubbleSort.c
#include <stdio.h>

void bubbleSort(int arr[], int count)
{
	int i=count, j;
	int temp;

	while(i>0)
	{
		for(j=0; j<i-1; j++)
		{
			if(arr[j] > arr[j+1])
			{
				temp = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = temp;
			}
		}
		i--;
	}
}

int main()
{
	int arr[] = {5, 4, 1, 3, 6};
	bubbleSort(arr, 5);
	int i;
	for(i=0; i<5; i++)
		printf("%4d", arr[i]);
}

File: bubbleSort.py
#!/usr/bin/env python

def bubbleSort(theSeq):

	n = len(theSeq)

	for i in xrange(n-1):
		swapCount = 0
		for j in xrange(n-1-i):
			if theSeq[j] > theSeq[j+1]:
				swapCount += 1
				theSeq[j], theSeq[j+1] = theSeq[j+1], theSeq[j]
		if not swapCount:
			print "Repeat count : %s"%(i+1)
			break



if __name__ == '__main__':
	testSeq = [3,4,1,100,90,76,1000]
	bubbleSort(testSeq)

	for item in testSeq:
		print item